### PR TITLE
Update model importer to load the dataset policies as resources.

### PR DESCRIPTION
**Pull request by ahoying**
* Update inventory storage to return resources with other type classes.

Thanks for opening a Pull Request!

Here's a handy checklist to ensure your PR goes smoothly.

- [x] I signed Google's [Contributor License Agreement](https://opensource.google.com/docs/cla/)
- [x] My code conforms to Google's [python style](https://google.github.io/styleguide/pyguide.html).
- [x] My PR at a minimum doesn't decrease unit-test coverage (if applicable).
- [x] My PR has been functionally tested.
- [x] All of the [unit-tests](http://forsetisecurity.org/docs/development/#executing-tests) still pass.
- [x] Running `pylint --rcfile=pylintrc` passes.

These guidelines and more can be found in our [contributing guidelines](https://github.com/GoogleCloudPlatform/forseti-security/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: google/cloud/forseti/services/model/importer/importer.py [hunk 1]
```diff
@@ -536,25 +544,37 @@ def _convert_appengine_resource(self, gae_resource):
         self.session.add(resource)
         self._add_to_cache(gae_resource, resource)
 
-    def _convert_dataset(self, dataset):
+    def _convert_dataset(self, dataset, policy):
```
Comment (jiyunyao) [inline google/cloud/forseti/services/model/importer/importer.py]: I don't think it is a good idea to deal with dataset and dataset_policy at the same time. It is a one off and temporary solution. I would suggest using the same flow as iam_policy.

File: google/cloud/forseti/services/model/importer/importer.py [hunk 2]
```diff
@@ -145,6 +144,10 @@ def run(self):
             'cloudsqlinstance'
             ]
 
+        bigquery_type_list = [
```
Comment (jiyunyao) [inline google/cloud/forseti/services/model/importer/importer.py]: If it is gcp resources, it should go to gcp_type_list

File: google/cloud/forseti/services/model/importer/importer.py [hunk 3]
```diff
@@ -175,6 +178,14 @@ def run(self):
                 self._store_resource(None, last_res_type)
                 self.session.flush()
 
+                for (policy, resource) in inventory.iter(
```
Comment (jiyunyao) [inline google/cloud/forseti/services/model/importer/importer.py]: the dataset policy should be treated similar to iam policy. in _store_dataset_policy, the dataset policy can be stored to resource table. When we decide how to deal with dataset policy, the changes will be limited to _store_dataset_policy

File: google/cloud/forseti/services/inventory/storage.py [hunk 4]
```diff
@@ -764,6 +775,17 @@ def iter(self,
                         Inventory.parent_key == p_key,
                         Inventory.parent_type == p_type,
                         parent_inventory.index == self.index.id)))
+        elif with_resource:
```
Comment (jiyunyao) [inline google/cloud/forseti/services/inventory/storage.py]: This is not necessary if dataset policy follow the policy processing flows like iam policy flow.

**Comment by ahoying** [comment]
PTAL.

**Comment by ahoying** [comment]
Lint errors fixed in https://github.com/GoogleCloudPlatform/forseti-security/pull/900.

**Pull request merged**